### DISCUSSION TITLE
Manifest generation for m2e-maven-runtime fails on Windows

### DISCUSSION THREAD

**Discussion by karypid** [eclipse-m2e/m2e-core#822] [General] [answered]
This is in continuation to [this mailing list thread](https://www.eclipse.org/lists/m2e-dev/msg02486.html).

I have just found out that the manifest generation fails on Windows with:

```
> mvn generate-sources -f m2e-maven-runtime -Pgenerate-osgi-metadata -Dtycho.mode=maven
[WARNING] org.eclipse.tycho.extras:tycho-pomless build extension will be replaced in a future version of Tycho by the new org.eclipse.tycho:tycho-build extension.
[WARNING] You can simply change your .mvn/extensions.xml to reference the new extension right now:
[WARNING] <extension>
[WARNING]    <groupId>org.eclipse.tycho</groupId>
[WARNING]    <artifactId>tycho-build</artifactId>
[WARNING]    <version>2.7.2</version>
[WARNING] </extension>
[INFO] Scanning for projects...
[ERROR] Internal error: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST C:\Users\user\devroot\wc.git\github.com\eclipse\m2e-core\m2e-maven-runtime\org.eclipse.m2e.archetype.common\META-INF\MANIFEST.MF: Manifest file not found -> [Help 1]
org.apache.maven.InternalErrorException: Internal error: org.eclipse.tycho.core.osgitools.OsgiManifestParserException: Exception parsing OSGi MANIFEST C:\Users\user\devroot\wc.git\github.com\eclipse\m2e-core\m2e-maven-runtime\org.eclipse.m2e.archetype.common\META-INF\MANIFEST.MF: Manifest file not found
```

Any chance someone knowledgable with Tycho can have a look and help out? Same command works in Linux with Maven 3.8 and Java 17...

**Comment by karypid** [comment, marked as answer]
I am attaching two files: `linux.txt` and `windows.txt`:

[linux.txt](https://github.com/eclipse-m2e/m2e-core/files/8985626/linux.txt)
[windows.txt](https://github.com/eclipse-m2e/m2e-core/files/8985627/windows.txt)

Both are the output of running the command on my Linux and Windows computers:

```
mvn generate-sources -f m2e-maven-runtime -Pgenerate-osgi-metadata -Dtycho.mode=maven -X -e > [OS].txt
```

The command works in Linux and fails on Windows. In the attached file, I have downgraded to Java 11 (just in case) on Windows but I get the same error when I try with Java 17 as well. It just fails on Windows.

If you search for the string `THIS IS WHERE WINDOWS DEVIATES FROM LINUX` in both, you will find the spot where Linux finishes parsing the pom and logs the reactor as it starts generation, whereas windows deviates and fails...

---

Argh! I've finally figured it out! **The culprit is powershell!**

If you're on Windows, either use CMD to run the command, OR if you're on powershell quote the `-D"tycho.mode"=maven` parameters properly as follows:

```
mvn generate-sources -f m2e-maven-runtime -Pgenerate-osgi-metadata -D"tycho.mode"=maven
```

**Reply by laeubi** [reply]
Can you try if it makes a difference using tycho 2.7.3 / 2.7.4-SNAPSHOT / 3.0.0-SNAPSHOT (see [tycho snapshot](https://github.com/eclipse/tycho/wiki#getting-tycho-snapshots) build)?

Also it seems you are using `Apache Maven 3.8.6` on windows and `Apache Maven 3.8.5` on linux, so maybe also a difference in maven version?

**Reply by karypid** [reply]
Downgraded to 3.8.5 and getting the same:

```
> mvn --version
Apache Maven 3.8.5 (3599d3414f046de2324203b78ddcf9b5e4388aa0)
Maven home: C:\Users\user\scoop\apps\maven\current
Java version: 17.0.3, vendor: Eclipse Adoptium, runtime: C:\Users\user\scoop\apps\temurin17-jdk\current
Default locale: en_GB, platform encoding: Cp1252
OS name: "windows 11", version: "10.0", arch: "amd64", family: "windows"
```

So tycho 2.7.2 yields this with all identical maven versions.

I added the plugin repository from from the instructions and tried with `-Dtycho-version="2.7.3" -U`, then `-Dtycho-version="2.7.4-SNAPSHOT" -U` and `-Dtycho-version="3.0.0-SNAPSHOT" -U` but other than downloading the new plugin versions the net result was the same again...

**Reply by karypid** [reply]
Argh! I've finally figured it out! **The culprit is powershell!**

If you're on Windows, either use CMD to run the command, OR if you're on powershell quote the `-D"tycho.mode"=maven` parameters properly as follows:

```
mvn generate-sources -f m2e-maven-runtime -Pgenerate-osgi-metadata -D"tycho.mode"=maven
```
